torch_np/_detail/_reductions.py

```diff
@@ -36,12 +36,22 @@ def count_nonzero(a, axis=None):
 
 def argmax(tensor, axis=None):
     axis = _util.allow_only_single_axis(axis)
+
+    if tensor.dtype == torch.bool:
+        # RuntimeError: "argmax_cpu" not implemented for 'Bool'
+        tensor = tensor.view(torch.int8)
```

Comment:
Is the equivalence guaranteed in pytorch?

Reply:
I would cast it to `uint8` to be 100% sure, but yeah, I believe it is guaranteed by some work @peterbell10 did some time ago.

Reply:
A cast is guaranteed to be 0 or 1, but a view could contain any byte value.

Reply:
Oh and of course the work I did doesn't apply to the `torch.compile` stack.

Reply:
In this case, if viewing with `uint8` we would get something that's non zero, so argmax would then work, but sure, let's do `to`, as the compiler should be able to optimise out the copy really

Reply:
IIRC argmax always returns the lowest index that's equal to the maximum, so the distinction does matter.